pipeline/pipeline.go

```diff
@@ -0,0 +1,111 @@
+package pipeline
+
+import "github.com/51Degrees/device-detection-go/v4/dd"
+
+type Pipeline struct {
+	logger                        logWrapper
+	dataFile                      string
+	licenceKey                    string
+	createTempDataCopy            bool
+	dataUpdateOnStartEnabled      bool
+	dataFileUrl                   string
+	dataFilePullEveryMs           int
+	isScheduledFilePullingEnabled bool
+	loggerEnabled                 bool
+	manager                       *dd.ResourceManager
+	config                        *dd.ConfigHash
+}
+
+// NewResultsHash creates a new ResultsHash object
+// uaCapacity is the initial capacity of the User-Agent cache
+// overridesCapacity is the initial capacity of the overrides cache
+func (p *Pipeline) NewResultsHash(uaCapacity uint32, overridesCapacity uint32) *dd.ResultsHash {
+	return dd.NewResultsHash(p.manager, uaCapacity, overridesCapacity)
+}
+
+// Run starts the pipeline
+func (p *Pipeline) Run() error {
```

Comment:
let's remove this method, the element or engine must have a method `Process` that takes `FlowData` as a parameter.  `FlowData` is just a container for evidence and intermediate input and output - a stateful computation results container.  We don't have `Pipeline` and `FlowData` in go - so just pass evidence as a parameter.  The data file update stuff should happen on construction of the Engine, see [JS implementation](https://github.com/51Degrees/device-detection-node/blob/main/fiftyone.devicedetection.onpremise/deviceDetectionOnPremise.js) of the engine, not when we run it - it's too late to update Data File here.